operation/push.go

```diff
@@ -5,27 +5,46 @@ import (
 	"fmt"
 	"io"
 	"os"
+	"time"
 
 	"gopkg.in/yaml.v3"
 
 	"github.com/cloudfoundry-community/go-cfclient/v3/client"
 	"github.com/cloudfoundry-community/go-cfclient/v3/resource"
 )
 
+type StrategyMode int
+
+const (
+	StrategyNone StrategyMode = iota
+	StrategyBlueGreen
+	StrategyRolling
+)
+
+func (sm StrategyMode) String() string {
+	return [...]string{"none", "blue-green", "rolling"}[sm]
```

Comment:
Directly accessing the slice in this manner may lead to an index out of range panic. Could you please use an alternative approach to avoid this issue

Reply:
```go
var strategyNames = map[StrategyMode]string{
	StrategyNone:      "none",
	StrategyBlueGreen: "blue-green",
	StrategyRolling:   "rolling",
}

func (sm StrategyMode) String() string {
	if name, found := strategyNames[sm]; found {
		return name
	}
	return "none"
}
```

Reply:
Thanks for your super quick response! 😊
Resolved.

Reply:
Moving the map inside the function fixes the threading issue, but creating a map seems wasteful/slower than just using a switch as used elsewhere in this module. For example https://github.com/cloudfoundry-community/go-cfclient/blob/fa1d959802a20798a6757b768ec59edf46fcdb33/resource/app.go#L92-L101

Reply:
Rethinking this function was called only to print error message for invalid strategy.
Ideally, since strategy is private field and we have setter WithStrategy we shouldn't let invalid strategy to pass through, removed the code and modified the WithStrategy setter.